### PR TITLE
[2.6] Add June k8s patches

### PR DESCRIPTION
**Pull request by dhruvmewada15**
Adds:

```
- 1.23.8-rancher1-1
- 1.22.11-rancher1-1	
- 1.21.14-rancher1-1	
- 1.20.15-rancher2-1
```

Linked Issues:
* https://github.com/rancher/rancher/issues/38004
* https://github.com/rancher/rancher/issues/37620

### CODE REVIEW

File: data/data.json [hunk 1]
```diff
@@ -9499,6 +9499,49 @@
    "aciGbpServerContainer": "noiro/gbp-server:5.1.1.0.1ae238a",
    "aciOpflexServerContainer": "noiro/opflex-server:5.1.1.0.1ae238a"
   },
+  "v1.21.14-rancher1-1": {
+   "etcd": "rancher/mirrored-coreos-etcd:v3.4.16-rancher1",
+   "alpine": "rancher/rke-tools:v0.1.80",
```
Comment (jakefhyde) [inline data/data.json]: @dhruvmewada15 Could you bump the `rke-tools` version to `v0.1.83` for all new versions? This is the latest version and should include fixes for issues such as https://github.com/rancher/rancher/issues/37639.

**Review by jakefhyde — CHANGES_REQUESTED** [review]

File: data/data.json (hunk 1 above)
Comment (dhruvmewada15) [reply data/data.json]: @jakefhyde Updated the `rke-tools` version to `v0.1.83`

**Review by jakefhyde — DISMISSED** [review]
LGTM

**Comment by dhruvmewada15** [comment]
Updated the PR with v1.20.15-rancher2 - which addresses https://github.com/rancher/rancher/issues/37620

File: rke/k8s_rke_system_images.go [hunk 2]
```diff
@@ -6614,6 +6614,50 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
 			WindowsPodInfraContainer:  "rancher/mirrored-pause:3.6",
 			Nodelocal:                 "rancher/mirrored-k8s-dns-node-cache:1.15.13",
 		},
+		// Enabled out of band post v2.6.5
```
Comment (snasovich) [inline rke/k8s_rke_system_images.go]: The current plan is to release June patches with v2.6.7, so let's update comment for this and newly added entries to:
```suggestion
		// Enabled with v2.6.7
```

The issues are still tracked in their own milestone, but likely will be updated to v2.6.7 too.

**Review by snasovich — DISMISSED** [review]
One minor comment about comments, otherwise LGTM. Approving in advance assuming the comment is addressed.

Waiting on https://github.com/rancher/hyperkube/pull/272

**Review by snasovich — APPROVED** [review]

**Review by rosskirkpat — APPROVED** [review]

**Comment by snasovich** [comment]
https://github.com/rancher/hyperkube/pull/272 is merged, new version tagged, and `hyperkube:v1.20.15-rancher2` has been successfully published to DockerHub. Merging this one.

**Pull request merged**